### ISSUE TITLE
Release 9.0.0

### ISSUE THREAD

**Issue by BasThomas** [Moya/Moya#1075]
After #762 is ready and merged, we should look at releasing `9.0.0`! 🎉🚀

- [x] Queues (#762)
- [x] `rx.request` / `reactive.request` (#1140, #1153, #1163)
- [x] Removing deprecated methods/types from 8.0.0  (#1162)
- [x] Check 8.0 -> 9.0 migration with an old project and check if the fix-its work as expected / there is need for a migration guide

**Comment by sunshinejr** [comment]
I think it is time to focus more on this release in the near future. There were multiple issues regarding our new introductions in 9.0.0 and Swift 4 is on the horizon. I'd love to have queues changes (#762) in, and if we can, `rx.request` & `reactive.request` from #963 and `sampleData` from #1126 as well. These are quite big breaking changes so I think we could try and wait for that, but we might want to just go without them if it takes too much time. What do you guys think?

**Comment by BasThomas** [comment]
Yeah, I think we should try to get 9.0.0 out of the door soonish, also to avoid having 10.0.0 being around the corner immediately.

I feel like the queue changes are first priority. `rx.request` and `reactive.request` are not that difficult to implement (I think?). As both are breaking changes, it would be nice to have them both in one release.

I think the `sampleData` is a little less of a priority, and we might want to put that into 10.0.0 if it takes too long.

**Comment by sunshinejr** [comment]
Exactly my point of view as well. Let's just try and finish queues, then pursue `rx.request` & `reactive.request` (which shouldn't be hard to implement, but if it turns out there is something in the way, we might want to skip it as well). If we can get `sampleData` PR in the meantime - great. If not, we may release 9.0.0 anyways.

**Comment by BasThomas** [comment]
I was thinking of a migration strategy to go from `RxMoyaProvider` to `MoyaProvider.rx` and from `ReactiveSwiftMoyaProvider` to `MoyaProvider.reactive`, but couldn't think of any where we can have a fix-it. Then I thought of something else. What we **can** do, is deprecate `RxMoyaProvider` and `ReactiveSwiftMoyaProvider`'s `init`s and make them private in 10.0.0, therefore having a more user-friendly migration as well as a transition with deprecation.

**Comment by sunshinejr** [comment]
Yeah, that's a good idea, let's do it 👍

**Comment by sunshinejr** [comment]
I'm thinking of merging `9.0.0-dev` into master after we merge queues & removing deprecations. What do you guys think?

**Comment by SD10** [comment]
@sunshinejr I think that's a good idea. #1126 SampleData still needs some time and I think there are enough changes in 9.0 as is. I recommend we move it to 10.0 to not hold up this release.

**Comment by sunshinejr** [comment]
Removed sample data from the todo-list and checked availability point since it has been recently merged. Only queues left (which probably needs a double-check only) and a final check for the entire release.

I also think it might be a good idea to do a beta release before the official one. Seems like quite a lot has changed! 😄

**Comment by sunshinejr** [comment]
Update! 🎉 
 
I've managed to merge changes from master & 9.0.0. I Must say it wasn't _that_ easy, but I think I've got everything in the end. I'm gonna make a PR in a sec, because it would be awesome to double check we got everything. Especially reactive providers with tests, since we 

Also, why alpha instead of beta? There is this PR #1147, it progressed really fast from zero to almost ready state. I'm not sure if we can squeeze this one in 9.0.0, we gotta think about it for sure, but making an alpha release gives us more flexibility.

**Comment by BasThomas** [comment]
+1 on the alpha.

**Comment by sunshinejr** [comment]
We can also think about adding Swift 4.0 support. From what I've seen we just need to change `NetworkLoggerPlugin` init, because it uses `fileprivate` method as a default parameter, thus Swift 4 compiler complains. Simple fix would be to use optional `output` parameter with `nil` as a default argument and in case `nil` is passed, use `NetworkLoggerPlugin.reversedPrint`. This way compiler shouldn't complain anymore 😄 

This might be a good idea just after an alpha merge since we probably don't want to mess with that PR anymore 😄

**Comment by BasThomas** [comment]
Seems like a great idea, especially when it would be that simple. No reason to make it another major version bump just for Swift 4.

**Comment by sunshinejr** [comment]
Okay everyone, alpha is merged. I'm gonna do a release in a bit 🎉

**Comment by BasThomas** [comment]
Thanks for all the hard work you've been putting into this, @sunshinejr! 💪

**Comment by sunshinejr** [comment]
9.0.0-alpha released! Thanks to everyone involved 💪💪💪

We can try to focus on the next release now 😄 Apart from bugs/patches we need to do, we can:
- Update `NetworkLoggerPlugin` to make Moya compatible with Swift 4.0
- If we update our codebase to support Swift 4.0, we shoud update our CI setup to build both 3.X version and 4.0.
- Add #1147 (+ documentation/tests).
- Update AccessTokenPlugin to accept `Basic` or `Bearer`. This should be covered by @yar1vn from #1172.

What do you think guys? Something to add, something to remove?

**Comment by SD10** [comment]
@sunshinejr When referring to the next release, are you talking about `9.0.0-beta` or `10.0.0`?

**Comment by sunshinejr** [comment]
Oh, I mean in 9.0.0 context - beta.1/alpha.2, depends how we feel about what we can still add to the pile.

**Comment by Dschee** [comment]
I'm happy #1147 is planned to be added, too – I think documentation is up-to-date, the only missing thing should be tests. I've got a question on the last bullet point of the initial post though:

> Check 8.0 -> 9.0 migration with an old project and check if the fix-its work as expected / there is need for a migration guide

First, do I understand it correctly, you were able to add Xcode fix-its to the code? I didn't know we can do this. If so, I should probably add some fix-its to #1147 as well. And second, #1147 does include a `MigrationGuides.md` file which could be extended. Also it should be probably linked to within the `Readme.md` or at least in the release notes.

**Comment by BasThomas** [comment]
On availability / fix-its: check https://github.com/Moya/Moya/pull/890. Example: https://github.com/Moya/Moya/issues/888#issuecomment-270004469

**Comment by Dschee** [comment]
@BasThomas Thanks for the pointers. I've tried to add fix-its but my approach failed. See https://github.com/Moya/Moya/pull/1147#issuecomment-318890623 for more details.

**Comment by sunshinejr** [comment]
We are closing in for the next release! 🎉  I think the last PR that is needed is #1241. We've come really far with this one! 

The only question I'm having is whether we should release `alpha.2` or `beta.1`? Do we have more breaking changes that we want to include in this release? 🤔

**Comment by SD10** [comment]
@sunshinejr Thanks for taking care of those unit tests 💪 I think we should go with `beta.1`. I don't have any more breaking changes personally and I don't think we should hold up the release for much longer. There's been more issues regarding Swift 4 recently and people looking to use some of the changes made in `9.0.0`.

**Comment by sunshinejr** [comment]
That's what I was thinking as well! We need to speed up with the process a little bit, we can add more breaking changes in the 10.0 from now on. Thanks for the input, @SD10!

So, we've got tests merged in (we've got a lot of improvement lately thanks to @phimage), I'm gonna release `beta.1` in a while! 🎉

**Comment by sunshinejr** [comment]
Beta 1 released! Thanks to everyone! 🎉 🎉 🎉 

Now it's time to make sure it is all working good before final release. 😄

**Comment by BasThomas** [comment]
Thanks for all the effort, everyone - and @sunshinejr in particular. 🙌

**Comment by SD10** [comment]
Thank you for all your hard-work @sunshinejr @BasThomas 🍺 🎉

**Comment by phimage** [comment]
Thank you all to check my PR and for the hard-work

> to make sure it is all working well

To achieve this I make a new PR about multipart part upload

**Comment by sunshinejr** [comment]
Alright everybody. After weeks of testing we got only one issue and it's fixed now by @SD10 at #1252. We should aim for a release! 🎉 I think that we can test it out till the end of this weekend and release it on Monday. At least that is a time I would like to have to test it out 😄. In the meantime we can check out documentation (especially migration) if everything is alright. 

What do you guys think? 😉

**Comment by BasThomas** [comment]
Seems like a great plan. Let's get this shipped!

**Comment by sunshinejr** [comment]
Alright, this is exciting! Doing final checks and releasing this bad boy in a few! 🎉 Gonna keep you guys posted!

**Comment by sunshinejr** [comment]
Aaaaaand it's out! 🎉 We can officially close this one, it was a fun ride! Thank you all, you are awesome 🥇 

<img width="564" alt="zrzut ekranu 2017-09-04 o 21 11 35" src="https://user-images.githubusercontent.com/5232779/30036686-d8adbbc4-91b5-11e7-94e7-467f7bb23d48.png">
